src/components/EventSubMenu.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { Icon } from '@iconify/react';
+import { NavLink } from 'react-router-dom';
+
+const eventSubMenu = [
+  {
+    name: 'Availability',
+    icon: 'fa-solid:user-clock',
+    isNavLink: true,
+    path: '/availability',
+  },
+  {
+    name: 'Share',
+    icon: 'fa-solid:share-alt',
+    isNavLink: false,
+  },
+  {
+    name: 'Edit',
+    icon: 'bx:bxs-edit',
+    isNavLink: false,
+  },
+  {
+    name: 'Messages',
+    icon: 'bx:bxs-message',
+    isNavLink: true,
+    path: '/messages',
+  },
+];
+export default function EventSubMenu() {
+  function handleIconClick(iconClicked) {
+    let iconName;
+    if (iconClicked.target.tagName === 'svg') {
+      iconName = iconClicked.target.parentElement.attributes.name.value;
+    } else if (iconClicked.target.tagName === 'path') {
+      iconName =
+        iconClicked.target.parentElement.parentElement.attributes.name.value;
+    }
```

Comment:
This logic seems unneccessary, you can just pass the `icon.name` to `handleIconClick` at line 66. Is there anything I'm missing?

Reply:
Changed it like you suggested, it works like a charm:D